crates/noirc_frontend/src/hir/def_collector/dc_mod.rs

```diff
@@ -172,7 +180,11 @@ impl<'a> ModCollector<'a> {
                 self.def_collector.def_map.modules[self.module_id.0].declare_struct(name, id);
 
             if let Err((first_def, second_def)) = result {
-                let err = DefCollectorErrorKind::DuplicateFunction { first_def, second_def };
+                let err = DefCollectorErrorKind::Duplicate {
+                    typ: "type definition".to_string(),
```

Comment:
Notice that in the old implementation error was DuplicateFunction, but it should be duplicate type definition !